File: .kiro/skills/md2html/scripts/convert.mjs
#!/usr/bin/env node
// md2html — Node engine (markdown-it + markdown-it-footnote)

import { readFileSync, writeFileSync } from 'fs';
import { basename, dirname, join, resolve } from 'path';
import { fileURLToPath } from 'url';
import markdownIt from 'markdown-it';
import footnote from 'markdown-it-footnote';

const __dirname = dirname(fileURLToPath(import.meta.url));

// Parse args
const args = process.argv.slice(2);
let input, theme = 'dark', output;
for (let i = 0; i < args.length; i++) {
  if (args[i] === '--theme') theme = args[++i];
  else if (args[i] === '--output') output = args[++i];
  else if (!input) input = args[i];
}
if (!input) { console.error('Usage: convert.mjs <input.md> [--theme dark|github|light] [--output path.html]'); process.exit(1); }

const mdText = readFileSync(resolve(input), 'utf-8');
const sharedCss = readFileSync(join(__dirname, 'themes.css'), 'utf-8');

// Count headings for TOC decision
const headings = [];
const md = markdownIt({ html: true, linkify: true, typographer: true }).use(footnote);

// Collect headings via ruler
md.core.ruler.push('collect_headings', (state) => {
  for (const token of state.tokens) {
    if (token.type === 'heading_open') {
      const level = parseInt(token.tag.slice(1));
      const inline = state.tokens[state.tokens.indexOf(token) + 1];
      const text = inline?.children?.map(t => t.content).join('') || '';
      const id = text.toLowerCase().replace(/[^\w]+/g, '-').replace(/(^-|-$)/g, '');
      token.attrSet('id', id);
      if (level >= 2 && level <= 3) headings.push({ level, text, id });
    }
  }
});

const htmlBody = md.render(mdText);

// Build TOC if 3+ headings
let toc = '';
if (headings.length >= 3) {
  const items = headings.map(h =>
    `<li style="padding-left:${(h.level - 2) * 0.8}rem"><a href="#${h.id}">${h.text}</a></li>`
  ).join('\n');
  toc = `<button class="toc-toggle" onclick="document.querySelector('.toc-sidebar').classList.toggle('open')" aria-label="Toggle contents">☰</button>
<nav class="toc-sidebar"><strong>Contents</strong><ul>${items}</ul></nav>
<div class="toc-overlay" onclick="document.querySelector('.toc-sidebar').classList.remove('open')"></div>`;
}

const title = basename(input, '.md').replace(/[-_]/g, ' ').replace(/\b\w/g, c => c.toUpperCase());
const outPath = output || input.replace(/\.md$/, '.html');

const html = `<!DOCTYPE html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1">
<title>${title}</title>
<style>${sharedCss}</style>
</head><body class="theme-${theme}">
${toc}${htmlBody}
<div class="footer">Generated with md2html · node engine · ${theme} theme</div>
</body></html>`;

writeFileSync(resolve(outPath), html, 'utf-8');
console.log(`✅ ${outPath} (node · ${theme} · ${headings.length} sections)`);

File: .kiro/skills/md2html/scripts/convert.py
#!/usr/bin/env python3
"""md2html — Python engine (markdown + pygments)."""

import argparse, pathlib, markdown
from markdown.extensions.codehilite import CodeHiliteExtension
from markdown.extensions.toc import TocExtension

SCRIPT_DIR = pathlib.Path(__file__).parent
HIGHLIGHT = {"dark": "monokai", "github": "default", "light": "default"}


def convert(md_path, theme="dark", output=None):
    md_text = pathlib.Path(md_path).read_text(encoding="utf-8")
    shared_css = (SCRIPT_DIR / "themes.css").read_text(encoding="utf-8")

    from pygments.formatters import HtmlFormatter
    pygments_css = HtmlFormatter(style=HIGHLIGHT[theme]).get_style_defs('.codehilite')

    md = markdown.Markdown(extensions=[
        "tables", "fenced_code", "footnotes", "attr_list", "def_list",
        CodeHiliteExtension(css_class="codehilite", guess_lang=True, linenums=False),
        TocExtension(marker="[TOC]", toc_depth="2-3"),
    ])
    html_body = md.convert(md_text)

    toc = ""
    if hasattr(md, "toc_tokens") and len(md.toc_tokens) >= 3:
        toc = f'''<button class="toc-toggle" onclick="document.querySelector('.toc-sidebar').classList.toggle('open')" aria-label="Toggle contents">☰</button>
<nav class="toc-sidebar"><strong>Contents</strong>{md.toc}</nav>
<div class="toc-overlay" onclick="document.querySelector('.toc-sidebar').classList.remove('open')"></div>'''

    title = pathlib.Path(md_path).stem.replace("-", " ").replace("_", " ").title()
    out_path = pathlib.Path(output) if output else pathlib.Path(md_path).with_suffix(".html")
    out_path.write_text(f"""<!DOCTYPE html>
<html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width,initial-scale=1">
<title>{title}</title>
<style>{shared_css}\n{pygments_css}</style>
</head><body class="theme-{theme}">
{toc}{html_body}
<div class="footer">Generated with md2html · python engine · {theme} theme</div>
</body></html>""", encoding="utf-8")
    print(f"✅ {out_path} (python · {theme} · {len(md.toc_tokens)} sections)")


if __name__ == "__main__":
    p = argparse.ArgumentParser(description="md2html — Python engine")
    p.add_argument("input", help="Markdown file path")
    p.add_argument("--theme", choices=["dark", "github", "light"], default="dark")
    p.add_argument("--output", help="Output HTML path")
    args = p.parse_args()
    convert(args.input, args.theme, args.output)
